cmd/mattermost/commands/channel_test.go

```diff
@@ -117,12 +117,63 @@ func TestCreateChannel(t *testing.T) {
 	defer th.TearDown()
 
 	id := model.NewId()
-	name := "name" + id
-
-	th.CheckCommand(t, "channel", "create", "--display_name", name, "--team", th.BasicTeam.Name, "--name", name)
-
-	name = name + "-private"
-	th.CheckCommand(t, "channel", "create", "--display_name", name, "--team", th.BasicTeam.Name, "--private", "--name", name)
+	commonName := "name" + id
+	team, _ := th.App.Srv().Store.Team().GetByName(th.BasicTeam.Name)
+
+	t.Run("should create public channel", func(t *testing.T) {
+		th.CheckCommand(t, "channel", "create", "--display_name", commonName, "--team", th.BasicTeam.Name, "--name", commonName)
+		channel, _ := th.App.Srv().Store.Channel().GetByName(team.Id, commonName, false)
+		assert.Equal(t, commonName, channel.Name)
+		assert.Equal(t, model.CHANNEL_OPEN, channel.Type)
+	})
+
+	t.Run("should create private channel", func(t *testing.T) {
+		name := commonName + "-private"
+		th.CheckCommand(t, "channel", "create", "--display_name", name, "--team", th.BasicTeam.Name, "--name", name, "--private")
+		channel, _ := th.App.Srv().Store.Channel().GetByName(team.Id, name, false)
+		assert.Equal(t, name, channel.Name)
+		assert.Equal(t, model.CHANNEL_PRIVATE, channel.Type)
+	})
+
+	t.Run("should create channel with header and purpose", func(t *testing.T) {
+		name := commonName + "-withhp"
+		th.CheckCommand(t, "channel", "create", "--display_name", name, "--team", th.BasicTeam.Name, "--name", name, "--header", "this is a header", "--purpose", "this is the purpose")
+		channel, _ := th.App.Srv().Store.Channel().GetByName(team.Id, name, false)
+		assert.Equal(t, name, channel.Name)
+		assert.Equal(t, model.CHANNEL_OPEN, channel.Type)
+		assert.Equal(t, "this is a header", channel.Header)
+		assert.Equal(t, "this is the purpose", channel.Purpose)
+	})
+
+	t.Run("should not create channel if name already exists on the same team", func(t *testing.T) {
+		output, err := th.RunCommandWithOutput(t, "channel", "create", "--display_name", commonName, "--team", th.BasicTeam.Name, "--name", commonName)
+		require.Error(t, err)
+		assert.True(t, strings.Contains(output, "A channel with that name already exists on the same team."))
```

Comment:
I think it would be more idiomatic if we would use [require.Contains](https://pkg.go.dev/github.com/stretchr/testify/require?tab=doc#Contains). We can use this every occurrence of  `assert.True(t, strings.Contains(...`

Reply:
Oh, cool! Didn't know about that one, I'm new to Go hehe

Reply:
done @isacikgoz